pkg/apis/core/validation/shoot.go

```diff
@@ -1992,10 +1998,11 @@ func ValidateSystemComponentWorkers(workers []core.Worker, kubernetesVersion str
 			continue
 		}
 
+		atLeastOnePoolWithAllowedSystemComponents = true
```

Comment:
I'm not sure that this is correct. We need the worker pool that has `sysemComponents.allow: true` to also have `worker.maximum` > `0`. Otherwise, I can create a shoot which has two pools: one with `worker.maximum` = `1` and `sysemComponents.allow`: `false`, and one with `worker.maximum` = `0` and `systemComponents.allow` = `true`.
I'm guessing that this is also what the `active` keyword in the `at least one active worker pool with systemComponents.allow=true needed` message refers to. 
This behaviour was also represented in one of the unit tests which was changed in this PR, however I don't think we should change it.

Note that `systemComponents.allow` is defaulted to true if it is not specified. That is why @Diaphteiros was able to create the shoot when he did not specify `systemComponents.allow`